Lazy load

Tominium opened the pull request:
Lazy loaded added to user list in dashboard

Tominium committed:
Lazy load

Tominium committed:
Added lazy load for user list. Does not work for list > 500

luukee commented:
@Tominium Since "[Does not work for list > 500](https://github.com/engagingnewsproject/misinfo-dashboard/pull/57/commits/990fbaef9b9b14173bff9ee713b5a8a8c485e910)" I don't think I should merge this. Lazy load should work regardless of how big the list is. What do you think?

Tominium commented:
Will look into it further

luukee committed:
Merge branch 'main' into lazy-loadv2

luukee committed:
Add user seeding script and update package.json accordingly

- Introduced a new script for seeding test users in the Firebase Authentication and Firestore emulators.
- Added a corresponding README.md file detailing prerequisites, usage, and troubleshooting for the seeding script.
- Updated package.json to include a new script command for seeding users.

luukee committed:
Update user seeding script and README to reflect changes in test user creation

- Adjusted the number of test users created from 13 to 12, with updated roles and email formats to include timestamps for uniqueness.
- Modified the README.md to accurately describe the new user roles and email structure.
- Enhanced the seed-users.js script to generate unique emails for regular, agency, and admin users, and added checks to prevent duplicate user creation in Firestore.
- Updated console logs for clarity on user processing and seeding summary.

luukee committed:
Refactor Users component.

- Updated JSDoc comments.
- Include total user count.
- New column for user index.
- Added deduplication logic when fetching more users to prevent duplicates in the user list.
- Role-based rendering for user details and actions based on user permissions.

luukee commented:
@Tominium Check out the [lazy-loadv2](https://github.com/engagingnewsproject/misinfo-dashboard/tree/lazy-loadv2) branch again and see if you can get lazy loading working properly for us. Let me know of any questions or issues you have.

_p.s. I added a script [see readme here](https://github.com/engagingnewsproject/misinfo-dashboard/blob/lazy-loadv2/scripts/README.md#L56) you can run to manually create users to test lazy loading with._
